File: app/src/main/java/com/malenalbc/superherotest/data/model/ResultContainer.java

package com.malenalbc.superherotest.data.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class ResultContainer<D extends Result> {

    @SerializedName("offset")
    @Expose
    public int offset;
    @SerializedName("limit")
    @Expose
    public int limit;
    @SerializedName("total")
    @Expose
    public int total;
    @SerializedName("count")
    @Expose
    public int count;
    @SerializedName("results")
    @Expose
    public List<D> results = new ArrayList<>();

    public List<D> getResults () {
        return results;
    }
}

File: data/src/main/java/com/malenalbc/stormcomics/data/model/ResultContainer.kt
package com.malenalbc.stormcomics.data.model

import com.google.gson.annotations.SerializedName


class ResultContainer<D : BaseResult>(
    @SerializedName("offset") var offset: Int = 0,
    @SerializedName("limit") var limit: Int = 0,
    @SerializedName("total") var total: Int = 0,
    @SerializedName("count") var count: Int = 0,
    @SerializedName("results") var results: List<D> = ArrayList()
)
